From 73a3b2738e6143699b19f9e57f27461bb497d75b Mon Sep 17 00:00:00 2001
From: Mengkejiergeli Ba <mengkejiergeli.ba@intel.com>
Date: Tue, 7 Nov 2023 16:20:10 +0800
Subject: [PATCH] msdk: Add device env in plugin dependencies

Add env vars GST_MSDK_DRM_DEVICE in plugin_add_dependencies to register
msdk plugins according to user's choice in a multi-gpu platform.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5615>
---
 sys/msdk/gstmsdk.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sys/msdk/gstmsdk.c b/sys/msdk/gstmsdk.c
index 9213be491..1542b7753 100644
--- a/sys/msdk/gstmsdk.c
+++ b/sys/msdk/gstmsdk.c
@@ -90,7 +90,8 @@ static void
 plugin_add_dependencies (GstPlugin * plugin)
 {
 #ifndef _WIN32
-  const gchar *env_vars[] = { "LIBVA_DRIVER_NAME", NULL };
+  const gchar *env_vars[] =
+      { "LIBVA_DRIVER_NAME", "GST_MSDK_DRM_DEVICE", NULL };
   const gchar *kernel_paths[] = { "/dev/dri", NULL };
   const gchar *kernel_names[] = { "card", "render", NULL };
 
-- 
2.40.1

